src/Internal/BalanceTx/BalanceTx.purs

```diff
@@ -365,36 +381,41 @@ runBalancer p = do
             }
             (Map.toUnfoldable p.utxos :: Array _)
 
-  mainLoop :: BalancerState -> BalanceTxM FinalizedTransaction
+  mainLoop :: BalancerState UnindexedTx -> BalanceTxM FinalizedTransaction
   mainLoop = worker <<< PrebalanceTx
     where
     worker :: BalancerStep -> BalanceTxM FinalizedTransaction
     worker (PrebalanceTx state) = do
       logBalancerState "Pre-balancing (Stage 1)" p.allUtxos state
       prebalanceTx state >>= runNextBalancerStep
-    worker (BalanceChangeAndMinFee s@{ transaction, minFee, leftoverUtxos }) =
+    worker (BalanceChangeAndMinFee state@{ transaction, minFee, leftoverUtxos }) =
       do
-        logBalancerState "Balancing change and fees (Stage 2)" p.allUtxos s
-        { transaction: balancedTx, minFee: newMinFee } <- evaluateTx s
+        logBalancerState "Balancing change and fees (Stage 2)" p.allUtxos state
+        { transaction: evaluatedTx, minFee: newMinFee } <- evaluateTx state
         case newMinFee <= minFee of
-          true ->
-            if (Set.isEmpty $ balancedTx ^. _body' <<< _inputs) then do
-              selectionState <-
-                performMultiAssetSelection p.strategy leftoverUtxos
-                  (lovelaceValueOf one)
-              runNextBalancerStep $ s
-                { transaction =
-                    balancedTx # _body' <<< _inputs %~
-                      Set.union (selectedInputs selectionState)
-                , leftoverUtxos =
-                    selectionState ^. _leftoverUtxos
-                }
-            else
-              logTransaction "Balanced transaction (Done)" p.allUtxos balancedTx
-                *> finalizeTransaction balancedTx p.allUtxos
+          true -> do
+            logTransaction "Balanced transaction (Done)" p.allUtxos
```

Comment:
The log message is the same as below in else branch, it could be changed to be more informative about balancer checking for an empty input set.